lib/java/CMakeLists.txt

```diff
@@ -17,14 +17,13 @@
 # under the License.
 #
 
-file(GLOB java_sources src/**/*.java)
-
 if(ANDROID)
     set(android_sources
         android/build.gradle
         android/settings.gradle
         android/src/main/AndroidManifest.xml
     )
+    file(GLOB java_sources src/**/*.java)
```

Comment:
It is better to add a cmake target that always runs the gradle build rather than having cmake attempt to enforce any dependency checking because it's likely to be wrong relative to the language build system.

Also,

https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil

Reply:
Done, no more file globbing!